lib/main_common.pm

```diff
@@ -1862,4 +1859,47 @@ sub load_sles4sap_tests {
     }
 }
 
+sub updates_is_applicable {
+    # we don't want live systems to run out of memory or virtual disk space.
+    # Applying updates on a live system would not be persistent anyway.
+    # Also, applying updates on BOOT_TO_SNAPSHOT is useless.
+    # Also, updates on INSTALLONLY do not match the meaning
+    return !get_var('INSTALLONLY') && !get_var('BOOT_TO_SNAPSHOT') && !get_var('DUALBOOT') && !get_var('UPGRADE') && !is_livesystem;
+}
+
+sub guiupdates_is_applicable {
+    return get_var("DESKTOP") =~ /gnome|kde|xfce|lxde/ && !check_var("FLAVOR", "Rescue-CD");
+}
+
+sub load_system_update_tests {
+    my (%args) = @_;
+    my $force_console = $args{force_console} || 0;
+
+    return if get_var('SYSTEM_UPDATED');
+    if (need_clear_repos() && !get_var('CLEAR_REPOS')) {
+        loadtest "update/zypper_clear_repos";
+        set_var('CLEAR_REPOS', 1);
+    }
+    loadtest "console/zypper_add_repos" if get_var('ZYPPER_ADD_REPOS');
+    return unless updates_is_applicable();
+    if (guiupdates_is_applicable() && !$force_console) {
+        loadtest "update/prepare_system_for_update_tests" if !is_sle;
+        if (check_var("DESKTOP", "kde")) {
+            loadtest "update/updates_packagekit_kde";
+            set_var('SYSTEM_UPDATED', 1);
+        }
+        elsif (x11tests_is_applicable()) {
+            unless (is_sle && is_staging) {
+                loadtest "update/updates_packagekit_gpk";
+                set_var('SYSTEM_UPDATED', 1);
+            }
+        }
+        loadtest "update/check_system_is_updated" if !is_sle;
+    }
+    elsif ($force_console && is_sle && !gnomestep_is_applicable()) {
+        loadtest "update/zypper_up";
+        set_var('SYSTEM_UPDATED', 1);
+    }
```

Comment:
Because the second `elsif` is the smaller part how about rewriting the if/elsif to

```perl
if ($force_console && is_sle && !gnomestep_is_applicable()) {
    loadtest 'update/zypper_up';
    set_var('SYSTEM_UPDATED', 1);
    return 1;
}
…
```

But the bigger question is: Have you understood by now if SLE and openSUSE need to differ here?